### PR TITLE
Code format check

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
CI stage to check the code style for gtest.

### CODE REVIEW

File: ci/azure-pipelines.yml [hunk 1]
```diff
@@ -23,8 +23,40 @@ resources:
...
+          demands:
+          - ucx_docker
+        container: fedora
+        steps:
+          - checkout: self
+            clean: true
+            fetchDepth: 100
+          - bash: |
+              set -x
+              BASE_SOURCEVERSION=$(git rev-parse HEAD^)
```
Comment (tvegas1) [inline ci/azure-pipelines.yml:45]: for my understanding, this would also work in case of multiple patches present in a given PR?

Comment (Alexey-Rivkin) [reply ci/azure-pipelines.yml:45]: Sure, whatever the commit contains.

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -23,8 +23,40 @@ resources:
...
+              echo "Checking code format on diff ${BASE_SOURCEVERSION}..${BUILD_SOURCEVERSION}"
+              PATCH=$(git-clang-format --diff ${BASE_SOURCEVERSION} ${BUILD_SOURCEVERSION})
```
Comment (tvegas1) [inline ci/azure-pipelines.yml]: Although it is being run under `$(System.DefaultWorkingDirectory)/test/gtest`, it seems that it still evaluates all the code in the diff. So I guess we need to add `./` to restrict to cpp code, like:
- `git-clang-format --diff ${BASE_SOURCEVERSION} ${BUILD_SOURCEVERSION} ./`

Comment (tvegas1) [reply ci/azure-pipelines.yml]: typo: the added `./` should be before `)`

Comment (Alexey-Rivkin) [reply ci/azure-pipelines.yml]: Sure! Thanks for noticing.

**Review by tvegas1 — APPROVED** [review]

File: ci/azure-pipelines.yml [hunk 3]
```diff
@@ -23,8 +23,40 @@ resources:
...
+                  exit
+              fi
+              echo "$PATCH" | git apply
+              if ! git diff --quiet --exit-code; then
+                set +x
+                echo "##vso[task.complete result=SucceededWithIssues;]DONE Code is not formatted correctly"
+              fi
+            condition: eq(variables['Build.Reason'], 'PullRequest')
+            displayName: Check gtest code format
+            workingDirectory: $(System.DefaultWorkingDirectory)/test/gtest
```
Comment (brminich) [inline ci/azure-pipelines.yml:58]: why do we do that for gtests only?

Comment (tvegas1) [reply ci/azure-pipelines.yml:58]: we currently use a basic clang-format for cpp under test/gtest, and do not have .clang-format for kernel (tabs, specific rules..), and user code yet.

**Review by brminich — APPROVED** [review]

**Pull request merged**